extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeCommentsExtractor.java

```diff
@@ -24,26 +14,37 @@
 import org.schabi.newpipe.extractor.linkhandler.ListLinkHandler;
 import org.schabi.newpipe.extractor.localization.Localization;
 import org.schabi.newpipe.extractor.utils.JsonUtils;
+import org.schabi.newpipe.extractor.utils.Utils;
 
-import com.grack.nanojson.JsonArray;
-import com.grack.nanojson.JsonObject;
-import com.grack.nanojson.JsonWriter;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getJsonPostResponse;
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getTextFromObject;
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.prepareDesktopJsonBuilder;
+import static org.schabi.newpipe.extractor.utils.Utils.isNullOrEmpty;
 
 public class YoutubeCommentsExtractor extends CommentsExtractor {
 
-    private JsonObject nextResponse;
+    /**
+     * The initial request's continuation token.
+     * Since we need to make two requests to get the comments,
+     */
+    private String initialToken;
+
+    /**
+     * Whether comments are disabled on video.
+     */
+    private boolean commentsDisabled = true;
 
     /**
-     * Caching mechanism and holder of the commentsDisabled value.
-     * <br/>
-     * Initial value = empty -> unknown if comments are disabled or not<br/>
-     * Some method calls {@link #findInitialCommentsToken()}
-     * -> value is set<br/>
-     * If the method or another one that is depending on disabled comments
-     * is now called again, the method execution can avoid unnecessary calls
+     * The second ajax <b>/next</b> response.
      */
-    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
-    private Optional<Boolean> optCommentsDisabled = Optional.empty();
+    private JsonObject ajaxJson;
```

Comment:
> I see an issue with this approach: if the client make a call to getCommentsCount() after loading some pages, it will use the comments count from latest next continuation response. YouTube provides only the comments section header for the first continuation.

I don't think this should be a problem since extractor objects shouldn't be reused? 

> This approach may drop however the ability to know extraction errors when trying to get comments count.

Yeah this is the issue with that approach, it would silently fail.

Reply:
Removed it  :+1:

I added it before since other extractors store it like that in a field (likely for caching reasons), but seeing there's no consistency in how it's done, I removed it.